File: references/Ai-Novel-main/frontend/src/lib/promptTaskCatalog.ts
import type { PromptStudioTask } from "../pages/promptStudio/types";
import { UI_COPY } from "./uiCopy";

export const PROMPT_TASK_CATALOG = [
  {
    key: "outline_generate",
    uiCopyKey: "outlineGenerate",
    label: UI_COPY.promptStudio.tasks.outlineGenerate,
  },
  {
    key: "chapter_generate",
    uiCopyKey: "chapterGenerate",
    label: UI_COPY.promptStudio.tasks.chapterGenerate,
  },
  {
    key: "plan_chapter",
    uiCopyKey: "planChapter",
    label: UI_COPY.promptStudio.tasks.planChapter,
  },
  {
    key: "post_edit",
    uiCopyKey: "postEdit",
    label: UI_COPY.promptStudio.tasks.postEdit,
  },
  {
    key: "content_optimize",
    uiCopyKey: "contentOptimize",
    label: UI_COPY.promptStudio.tasks.contentOptimize,
  },
  {
    key: "chapter_analyze",
    uiCopyKey: "chapterAnalyze",
    label: UI_COPY.promptStudio.tasks.chapterAnalyze,
  },
  {
    key: "chapter_rewrite",
    uiCopyKey: "chapterRewrite",
    label: UI_COPY.promptStudio.tasks.chapterRewrite,
  },
] as const satisfies ReadonlyArray<{
  key: string;
  uiCopyKey: keyof typeof UI_COPY.promptStudio.tasks;
  label: string;
}>;

export const PROMPT_STUDIO_TASKS: PromptStudioTask[] = PROMPT_TASK_CATALOG.map((item) => ({
  key: item.key,
  label: item.label,
}));

export const PROMPT_TASK_KEYS = PROMPT_STUDIO_TASKS.map((item) => item.key);

File: references/Ai-Novel-main/backend/app/services/prompt_task_catalog.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True, slots=True)
class PromptTaskCatalogItem:
    key: str
    ui_copy_key: str
    e2e_specs: tuple[str, ...]


PROMPT_TASK_CATALOG: tuple[PromptTaskCatalogItem, ...] = (
    PromptTaskCatalogItem(
        key="outline_generate",
        ui_copy_key="outlineGenerate",
        e2e_specs=("test/specs/api/prompt-task-reachability.contract.spec.ts",),
    ),
    PromptTaskCatalogItem(
        key="chapter_generate",
        ui_copy_key="chapterGenerate",
        e2e_specs=("test/specs/api/prompt-task-reachability.contract.spec.ts",),
    ),
    PromptTaskCatalogItem(
        key="plan_chapter",
        ui_copy_key="planChapter",
        e2e_specs=("test/specs/api/prompt-task-reachability.contract.spec.ts",),
    ),
    PromptTaskCatalogItem(
        key="post_edit",
        ui_copy_key="postEdit",
        e2e_specs=("test/specs/api/prompt-task-reachability.contract.spec.ts",),
    ),
    PromptTaskCatalogItem(
        key="content_optimize",
        ui_copy_key="contentOptimize",
        e2e_specs=("test/specs/api/prompt-task-reachability.contract.spec.ts",),
    ),
    PromptTaskCatalogItem(
        key="chapter_analyze",
        ui_copy_key="chapterAnalyze",
        e2e_specs=("test/specs/api/prompt-task-reachability.contract.spec.ts",),
    ),
    PromptTaskCatalogItem(
        key="chapter_rewrite",
        ui_copy_key="chapterRewrite",
        e2e_specs=("test/specs/api/prompt-task-reachability.contract.spec.ts",),
    ),
)

PROMPT_TASK_KEYS: tuple[str, ...] = tuple(item.key for item in PROMPT_TASK_CATALOG)
PROMPT_TASK_SET: frozenset[str] = frozenset(PROMPT_TASK_KEYS)


def is_supported_prompt_task(task: str) -> bool:
    return str(task or "").strip() in PROMPT_TASK_SET
